### ISSUE TITLE
Building enable with setuptools > 9.1 will fail if the requirements are not already installed

### ISSUE THREAD

**Issue by itziakos** [enthought/enable#179]
This has been reproduced on ubuntu 12.04 64 bit with system python 2.7.3 running on a virtualenv.

**Comment by itziakos** [comment]
```python
(penv)enthought@simphony-base:~/enable$ python setup.py develop
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/local/include/python2.7 is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path  is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/X11R6/lib64 is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/X11R6/lib is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/X11/lib64 is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/X11/lib is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/lib64 is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/X11R6/include is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/X11/include is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:576: UserWarning: Specified path /usr/lib/X11/include is invalid.
  warnings.warn('Specified path %s is invalid.' % d)
abort: there is no Mercurial repository here (.hg not found)
running build_src
build_src
building library "freetype2_src" sources
  adding 'kiva/agg/freetype2/src/autofit/autofit.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftbase.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftsystem.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftinit.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftglyph.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftmm.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftbdf.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftbbox.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftdebug.c' to sources.
  adding 'kiva/agg/freetype2/src/base/ftxf86.c' to sources.
  adding 'kiva/agg/freetype2/src/base/fttype1.c' to sources.
  adding 'kiva/agg/freetype2/src/bdf/bdf.c' to sources.
  adding 'kiva/agg/freetype2/src/cff/cff.c' to sources.
  adding 'kiva/agg/freetype2/src/cid/type1cid.c' to sources.
  adding 'kiva/agg/freetype2/src/lzw/ftlzw.c' to sources.
  adding 'kiva/agg/freetype2/src/pcf/pcf.c' to sources.
  adding 'kiva/agg/freetype2/src/pfr/pfr.c' to sources.
  adding 'kiva/agg/freetype2/src/psaux/psaux.c' to sources.
  adding 'kiva/agg/freetype2/src/pshinter/pshinter.c' to sources.
  adding 'kiva/agg/freetype2/src/psnames/psnames.c' to sources.
  adding 'kiva/agg/freetype2/src/raster/raster.c' to sources.
  adding 'kiva/agg/freetype2/src/sfnt/sfnt.c' to sources.
  adding 'kiva/agg/freetype2/src/smooth/smooth.c' to sources.
  adding 'kiva/agg/freetype2/src/truetype/truetype.c' to sources.
  adding 'kiva/agg/freetype2/src/type1/type1.c' to sources.
  adding 'kiva/agg/freetype2/src/type42/type42.c' to sources.
  adding 'kiva/agg/freetype2/src/winfonts/winfnt.c' to sources.
  adding 'kiva/agg/freetype2/src/gzip/ftgzip.c' to sources.
building library "agg24_src" sources
building library "kiva_src" sources
building extension "kiva.agg._agg" sources
building extension "kiva.agg._plat_support" sources
building data_files sources
build_src: building npy-pkg config files
running build_clib
customize UnixCCompiler
customize UnixCCompiler using build_clib
running build_ext
build_clib already run, it is too late to ensure in-place build of build_clib
customize UnixCCompiler
customize UnixCCompiler using build_ext
extending extension 'kiva.agg._agg' defined_macros with [('FT2_BUILD_LIBRARY', None)]
customize UnixCCompiler
customize UnixCCompiler using build_ext
running develop
running build_scripts
running egg_info
running build_src
build_src
building library "freetype2_src" sources
building library "agg24_src" sources
building library "kiva_src" sources
building extension "kiva.agg._agg" sources
building extension "kiva.agg._plat_support" sources
building data_files sources
build_src: building npy-pkg config files
writing requirements to enable.egg-info/requires.txt
writing enable.egg-info/PKG-INFO
writing top-level names to enable.egg-info/top_level.txt
writing dependency_links to enable.egg-info/dependency_links.txt
package init file 'enable/savage/trait_defs/ui/wx/data/__init__.py' not found (or not a regular file)
reading manifest file 'enable.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'chaco/tests/data/PngSuite/*.png'
warning: no files found matching 'chaco/tests/data/PngSuite/LICENSE.txt'
writing manifest file 'enable.egg-info/SOURCES.txt'
running build_ext
build_clib already run, it is too late to ensure in-place build of build_clib
customize UnixCCompiler
customize UnixCCompiler using build_ext
extending extension 'kiva.agg._agg' defined_macros with [('FT2_BUILD_LIBRARY', None)]
customize UnixCCompiler
customize UnixCCompiler using build_ext
Creating /home/enthought/enable/penv/lib/python2.7/site-packages/enable.egg-link (link to .)
enable 4.6.0.dev17 is already the active version in easy-install.pth

Installed /home/enthought/enable
Processing dependencies for enable==4.6.0.dev17
Searching for traits
Reading https://pypi.python.org/simple/traits/
Best match: traits 4.5.0
Downloading https://pypi.python.org/packages/source/t/traits/traits-4.5.0.tar.gz#md5=3ad558eebaedc63c29c80183c0371d2f
Processing traits-4.5.0.tar.gz
Writing /tmp/easy_install-kzrMv9/traits-4.5.0/setup.cfg
Running traits-4.5.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-kzrMv9/traits-4.5.0/egg-dist-tmp-koCiET
Traceback (most recent call last):
  File "setup.py", line 255, in <module>
    **config)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/core.py", line 169, in setup
    return old_setup(**new_attr)
  File "/usr/lib/python2.7/distutils/core.py", line 152, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/command/develop.py", line 32, in run
    self.install_for_development()
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/numpy/distutils/command/develop.py", line 17, in install_for_development
    old_develop.install_for_development(self)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/command/develop.py", line 132, in install_for_development
    self.process_distribution(None, self.dist, not self.no_deps)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 709, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 800, in resolve
    dist = best[req.key] = env.best_match(req, ws, installer)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1050, in best_match
    return self.obtain(req, installer)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1062, in obtain
    return installer(requirement)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 629, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 659, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 844, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1072, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1058, in run_setup
    run_setup(setup_script, args)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 227, in run_setup
    with setup_context(setup_dir):
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 193, in setup_context
    yield
  File "/usr/lib/python2.7/contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 152, in save_modules
    yield saved
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 126, in __exit__
    self._saved = UnpickleableException.dump(type, exc)
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
  File "/home/enthought/enable/penv/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 110, in dump
    return cls.dump(cls, cls(repr(exc)))
MemoryError
```

**Comment by itziakos** [comment]
My first gues is that there is a bud interaction between setuptools and numpy.distutils

**Comment by itziakos** [comment]
fixed by #182
